File: ruby/prime-factors/prime_factors.rb
# PrimeFactos comment
module PrimeFactors
  def self.for(num) 
    n = 2
    factors = []
    while num != 1
      if (num % n).zero?
        factors << n
        num /= n
      else
        n += 1
      end
    end
    factors
  end
end

File: python/prime-factors/prime_factors.py
def __version__():
    VERSION = 1


def factors(value):
    if value == 1:
        return []
    else:
        n = 2
        factors = []
        while value != 1:
            if value % n == 0:
                factors.append(n)
                value = value / n
            else:
                n += 1
        return factors
